### PR TITLE
Add deployment instructions for generic servers

### PR DESCRIPTION
**Pull request by JackWilb**
Adds instructions for building and deploying the client and the API. 

TODO:
- [x] Add example of nginx reverse proxy
- [x] Add a section for deploying on PaaS vendors (opened #21 for Heroku)

### CODE REVIEW

File: generic_deployment.rst [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+Since all of these services should be deployed onto separate servers/vms the
+easiest way to handle accessing them is through DNS. If you have a domain, you
+can set subdomains for each program in the stack. For example:
+
+* db for the ArangoDB instance
+* api for the ``multinet-server`` application
+* @ and www for the ``multinet-client`` application
+
+There may be some additional apps that you can host on the same domain. For
+example, we host MultiLink and MultiMatrix on subdomains of our main domain.
```
Comment (AlmightyYakob) [inline generic_deployment.rst:79]: I think it's a bit misleading to say that these services are supposed to be deployed onto separate servers/vms/etc. They can be, but it's not required. I also think these examples could be a bit more clear:

```suggestion
Handling DNS
------------

All of these services can be deployed onto separate servers/vms. If this is how
you want to deploy them, and you have a domain, you can assign a subdomain (through your DNS provider) for each program in the stack. For a given example domain ``domain.com``, you could organize the services as follows:

* domain.com ``multinet-client`` application
* db.domain.com for the ArangoDB instance
* api.domain.com for the ``multinet-server`` application

There may be some additional apps that you can host on the same domain. For
example, we host MultiLink and MultiMatrix on subdomains of our main domain.
```

In this example, I removed the `@` and `www` from the prefix for the client. I don't think the `www` is necessary, what is the `@` supposed to represent?

**Review by AlmightyYakob — CHANGES_REQUESTED** [review]
Great job with this, just a few suggestions

File: generic_deployment.rst (hunk 1 above)
Comment (JackWilb) [reply generic_deployment.rst:79]: `@` means apex in the DNS spec (e.g. multinet.app, so no subdomain)

`www` isn't absolutely necessary, but I feel like some people still type www.example.com in full. If you don't have that subdomain pointing at the IP for the client, it won't resolve.

I'll take most of your suggestions from above and add in the www

Comment (JackWilb) [reply generic_deployment.rst:79]: See 6a5c862

File: generic_deployment.rst [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+1. Clone the repository: ``git clone https://github.com/multinet-app/multinet-client``.
+2. Move into the repository: ``cd multinet-client``.
+3. Install the project dependencies: ``yarn install``.
+4. Build the production app: ``yarn build``.
+
+At this point, you have a few options for how to proceed. The most recommended
+way to continue is to serve these files using a production webserver such as
+`nginx <https://www.nginx.com/>`_ or `apache webserver <https://httpd.apache.org/>`_.
+See the docs for these projects for details on how to do that.
```
[the remark this answers is not in the document]
Comment (JackWilb) [reply generic_deployment.rst:65]: See e41bb4c

File: generic_deployment.rst [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+   an additional flag, for example: ``pipenv run gunicorn --bind unix:./socket.sock multinet.app:app``.
+   The socket will be located in the ``multinet-server`` directory.
+
+``multinet-client``
+-------------------
+
+1. Clone the repository: ``git clone https://github.com/multinet-app/multinet-client``.
+2. Move into the repository: ``cd multinet-client``.
+3. Install the project dependencies: ``yarn install``.
+4. Copy .env.default to .env and configure the ``VUE_APP_MULTINET_HOST``.
```
Comment (AlmightyYakob) [inline generic_deployment.rst:60]:
```suggestion
4. Copy .env.default to .env and configure ``VUE_APP_MULTINET_HOST`` to point to the address the server is running on (e.g. http://127.0.0.1:8000).
```

**Review by waxlamp — COMMENTED** [review]
This looks great. Please follow Jake's approval/requests in getting this merged.

**Review by AlmightyYakob — APPROVED** [review]
:+1:

**Review by AlmightyYakob — APPROVED** [review]

**Pull request merged**